### PR TITLE
feat(vue): add private Vue plugin

### PR DESCRIPTION
**Pull request by jaysoo**
This PR adds an empty vue plugin to start adding generators and executors to. It is marked as `private` for now so we don't publish until it is feature complete. The usual targets are setup (lint, test, build).

Also adds CODEOWNERS entry for Vue. For now it is the JS maintainers since Katerina, Nicholas, and myself are the same people working on Vue. We can change that later if needed.

### CODE REVIEW

File: packages/vue/package.json [hunk 1]
```diff
@@ -0,0 +1,44 @@
+{
+  "name": "@nx/vue",
+  "version": "0.0.1",
+  "private": true,
```
Comment (jaysoo) [inline packages/vue/package.json:4]: This will prevent `@nx/vue` from being published (except to local registry).

**Review by mandarini — APPROVED** [review]

File: CODEOWNERS [hunk 2]
```diff
@@ -54,6 +54,12 @@ pnpm-lock.yaml @nrwl/nx-core-reviewers
 /packages/react-native/** @nrwl/nx-react-reviewers
 /e2e/react-native/** @nrwl/nx-react-reviewers
 
+# Vue
+/docs/generated/packages/vue/** @nrwl/nx-js-reviewers @nrwl/nx-docs-reviewers
+/docs/shared/packages/vue/** @nrwl/nx-js-reviewers @nrwl/nx-docs-reviewers
+/packages/vue/** @nrwl/nx-js-reviewers
+/e2e/vue/** @nrwl/nx-js-reviewers
```
Comment (AgentEnder) [inline CODEOWNERS:61]: Can we go ahead and make this a separate team? I can create it if you update the code

File: packages/vue/package.json [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  "typings": "./index.d.ts",
+  "author": "Victor Savkin",
+  "license": "MIT",
+  "bugs": {
+    "url": "https://github.com/nrwl/nx/issues"
+  },
+  "homepage": "https://nx.dev",
+  "schematics": "./generators.json",
+  "builders": "./executors.json",
+  "ng-update": {
```
Comment (JamesHenry) [inline packages/vue/package.json:27]: I don't think we should use the Angular CLI compat syntax here, it won't ever be used in an Angular CLI workspace context, right?

```suggestion
  "nx-migrations": {
```

File: packages/vue/package.json [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+  "main": "./index",
+  "typings": "./index.d.ts",
+  "author": "Victor Savkin",
+  "license": "MIT",
+  "bugs": {
+    "url": "https://github.com/nrwl/nx/issues"
+  },
+  "homepage": "https://nx.dev",
+  "schematics": "./generators.json",
+  "builders": "./executors.json",
```
Comment (JamesHenry) [inline packages/vue/package.json:26]:
```suggestion
  "executors": "./executors.json",
```

Building on my comment about Angular CLI, do we really want `"builders"` here at all on a package that will never be used in an Angular CLI workspace?

File: packages/vue/package.json [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+  ],
+  "main": "./index",
+  "typings": "./index.d.ts",
+  "author": "Victor Savkin",
+  "license": "MIT",
+  "bugs": {
+    "url": "https://github.com/nrwl/nx/issues"
+  },
+  "homepage": "https://nx.dev",
+  "schematics": "./generators.json",
```
Comment (JamesHenry) [inline packages/vue/package.json:25]:
```suggestion
  "generators": "./generators.json",
```

Building on my comment about Angular CLI, do we really want "generators" here at all on a package that will never be used in an Angular CLI workspace?

File: packages/vue/executors.json [hunk 6]
```diff
@@ -0,0 +1,4 @@
+{
+  "builders": {},
```
Comment (JamesHenry) [inline packages/vue/executors.json:2]: Building on my comment about Angular CLI, do we really want "builders" here on a package that will never be used in an Angular CLI workspace?

**Review by JamesHenry — CHANGES_REQUESTED** [review]
Some comments all related to Angular CLI config

File: packages/vue/generators.json [hunk 7]
```diff
@@ -0,0 +1,6 @@
+{
+  "name": "Nx Vue",
+  "version": "0.1",
+  "schematics": {},
```
Comment (JamesHenry) [inline packages/vue/generators.json:4]:
```suggestion
```

Sorry missed this one before, same Angular CLI point

File: packages/vue/.eslintrc.json [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+    },
+    {
+      "files": ["./package.json"],
+      "parser": "jsonc-eslint-parser",
+      "rules": {
+        "@nx/dependency-checks": [
+          "error",
+          {
+            "buildTargets": ["build-base"],
+            "ignoredDependencies": ["nx", "typescript"]
```
Comment (meeroslav) [inline packages/vue/.eslintrc.json:32]: Why is Nx ignored? Shouldn't it be `peerDependency`?

File: tsconfig.base.json [hunk 9]
```diff
@@ -101,6 +101,7 @@
       "@nx/typedoc-theme": ["typedoc-theme/src/index.ts"],
       "@nx/vite": ["packages/vite"],
       "@nx/vite/*": ["packages/vite/*"],
+      "@nx/vue": ["packages/vue/src/index.ts"],
```
Comment (meeroslav) [inline tsconfig.base.json:104]: This path is incorrect. There is no `src/index.ts`.
```suggestion
      "@nx/vue": ["packages/vue"],
```

**Review by meeroslav — DISMISSED** [review]
Left two minor notes

**Review by JamesHenry — APPROVED** [review]

**Review by AgentEnder — APPROVED** [review]

**Review by FrozenPandaz — APPROVED** [review]

**Pull request merged**